pyensign/connection.py

```diff
@@ -193,8 +193,8 @@ async def publish(self, topic, events, on_ack=None, on_nack=None):
                 done_callback=lambda: self.publishers.pop(topic_hash, None),
             )
 
-        # Create a concurrent task to queue the events from the user
-        await self.pool.schedule(publisher.queue_events(events))
+        # Queue all the events to be published
+        await publisher.queue_events(events)
```

Comment:
Note that this makes the publish behavior a bit different because it blocks user code until all the events in the iterator are queued (although does not block the routine that writes the events to the stream), but I think it does make it a bit more obvious to the user which events have been queued.

90% of the Ensign publishers I've seen so far publish one or a few events at a time so I think this more in line with how people are using Ensign.